### ISSUE TITLE
Filter function not working with list of channel without group and index(same files works fine for older version of asamMdf)

### ISSUE THREAD

**Issue by vignesRectifier** [danielhrisca/asammdf#456]
# Python version
```
# Code 
'python=3.7.9 (default, Aug 31 2020, 17:10:11) [MSC v.1916 64 bit (AMD64)]'
'os=Windows-10-10.0.18362-SP0'
'numpy=1.18.5'
'asammdf=6.1.2'


  ## MDF version
  4.00

  ## Code snippet
self.__objMdf__.filter([list of signal names])
  
  ## Traceback
MdfException: Multiple occurances for channel "MO_Fahrpedalrohwert_01": ((4, 1), (6, 1)). Provide both "group" and "index" arguments to select another data group
  
# Description
Issue in filter function from current version which is working in old version

_Please describe the issue here._
Mdf object filter function works with above code snippet for version 5.21.0 but not with the latest(6.1.2).
Above mentioned channel is available only once.

Even the example code is giving the issue.

from asammdf import MDF, Signal
import numpy as np
t = np.arange(5)
s = np.ones(5)
mdf = MDF()
for i in range(4):
     sigs = [Signal(s*(i*10+j), t, name='SIG') for j in range(1,4)]
     mdf.append(sigs)

filtered = mdf.filter(['SIG', ('SIG', 3, 1), ['SIG', 2], (None, 1, 2)])
for gp_nr, ch_nr in filtered.channels_db['SIG']:
    print(filtered.get(group=gp_nr, index=ch_nr))


MdfException: Multiple occurances for channel "SIG": ((0, 1), (0, 2), (0, 3), (1, 1), (1, 2), (1, 3), (2, 1), (2, 2), (2, 3), (3, 1), (3, 2), (3, 3)). Provide both "group" and "index" arguments to select another data group

**Comment by danielhrisca** [comment]
Starting with 6.0.0 you need to specify exactly the entry (see #426)

**Comment by JulienGrv** [comment]
Your channel was found multiple times in your measurement. Calling get() would in the past return you the first entry found. This behavior was ambiguous and error-prone so we removed it. Now it is up to the user to fix the ambiguity by using the function whereis() and selecting the correct occurrence.

**Comment by vignesRectifier** [comment]
Thank you for your reply.. is it possible to get all the signals in that name by renaming automatically with index name append to the channel name?

**Comment by JulienGrv** [comment]
You can get all the signals like this:

```python
from pathlib import Path

from asammdf import MDF

path = Path("my_mdf.mf4")

mdf = MDF(path)

occurrences = mdf.whereis("my_signal")

signals = mdf.get(
    ((None, gp_idx, cn_idx) for gp_idx, cn_idx in occurrences)
)
```

Then you can rename them if you want by changing the attributes signal.name

**Comment by JulienGrv** [comment]
@vignesRectifier there was an error in my previous comment that I fixed. It's mdf.select and not mdf.get that must be used here

**Comment by vignesRectifier** [comment]
@JulienGrv  Thank you.. but we need mdf object instead of signal list for a single device. Multiple device logging to same file so we need to filter mdf object for a single device

**Comment by JulienGrv** [comment]
You can do this as well, the whereis function as source_name and source_path options to filter accordingly

**Comment by JulienGrv** [comment]
If you want to filter the MDF object, there is no feature for this right now do it's a feature request. Please open a new issue to ask for this feature and explain a bit what you need

**Comment by JulienGrv** [comment]
OK sorry, I understand you want to use the filter function and not select. This you can do it but you have to use whereis before to filter the multiple occurrences that you will then pass to the filter function. So just replace select with filter in my code snippet above.
